tests/MongoDB.Driver.Tests/Linq/IntegrationTestBase.cs

```diff
@@ -16,38 +16,58 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
 using MongoDB.Bson.Serialization.Attributes;
-using Xunit;
 
 namespace MongoDB.Driver.Tests.Linq
 {
     public abstract class IntegrationTestBase
     {
         protected static IMongoCollection<Root> __collection;
         protected static IMongoCollection<Other> __otherCollection;
+        protected static IMongoCollection<Root> __customCollection;
+        protected static List<Root> __customDocuments;
+
         private static ConcurrentDictionary<Type, bool> __oneTimeSetupTracker = new ConcurrentDictionary<Type, bool>();
 
         protected IntegrationTestBase()
         {
             __oneTimeSetupTracker.GetOrAdd(GetType(), OneTimeSetup); // run OneTimeSetup once per subclass
         }
 
+        protected virtual void FillCustomDocuments(List<Root> customDocuments)
+        {
+        }
+
         private bool OneTimeSetup(Type type)
         {
             var client = DriverTestConfiguration.Client;
             var db = client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
             __collection = db.GetCollection<Root>(DriverTestConfiguration.CollectionNamespace.CollectionName);
             __otherCollection = db.GetCollection<Other>(DriverTestConfiguration.CollectionNamespace.CollectionName + "_other");
+            __customCollection = db.GetCollection<Root>(DriverTestConfiguration.CollectionNamespace.CollectionName + "_custom");
```

Comment:
I've moved this line from here since we have several inheritors of this class, but we use this collection only in one of them. So, I decided that it's better to initialize this collection on-demand only: there is overload for `FillCustomDocuments` => initialize this collection, otherwise - no.

But I don't mind to have your implementation.